### PR TITLE
Feat/bulk/all create job

### PR DESCRIPTION
**Pull request by gweiying**
## Context

This PR adds logic to create and send out jobs to the QR code generation Lambda via SQS. 

The logic is as follows:
- After the url mappings have been created, url mappings are sent to the JobController
- JobController creates a Job,
- Job Controller batches the url mappings into several JobItems belonging to that Job 
  - for batch size of 100, 919 mappings will be split into 10 JobItems belonging to 1 Job, 9 JobItems with 100 mappings each and 1 JobItem with 19
- JobController will start these JobItems by sending it via SQS to QR code generation Lambda 

### SQS authentication
We have added security rules on the SQS queue so that only the EB IAM role is able to send incoming messages to the queue.

We experimented with VPC endpoints to secure the traffic between EB and SQS, but the VPC endpoint caused the EB deployments to fail. See https://stackoverflow.com/questions/55458492/create-aws-vpc-endpoint-for-sqs. After debugging for hours, we realised it was simpler to use the NAT gateway already attached to our private VPC security group to allow for outbound internet connections to the SQS queue. 

### Local development
Currently this feature does not work locally as there's no local stack set up yet. This is a lower priority task and will be tackled at the end.

## Tests

_What tests should be run to confirm functionality?_

- Deploy to staging,
- On Health staging (not configured on Edu and Go yet), create a bulk upload via Postman API
- You should be able to see the following
  - New job and job items should be created on health-staging DB jobs table and job_items table. (Check the job uuid, you'll need it for later)
  - SQS health-staging-bulk-qrcode-generate-start should receive the job item messages
  - Lambda health-staging-bulk-qrcode-generation should be triggered (can see invocations via Cloudwatch)
  - S3 bucket should have the newly generated files stored under job uuid from the DB table

## Deploy Notes

_Notes regarding deployment of the contained body of work. These should note any
new dependencies, new scripts, etc._

**New environment variables**:
- `SQS_BULK_QRCODE_GENERATE_START_URL`
- `SQS_TIMEOUT`
- `SQS_REGION`
- `BULK_QR_CODE_BATCH_SIZE`

### CODE REVIEW

**Review by halfwhole — APPROVED** [review]
LGTM, staging setup works really nicely! few nits below:

File: src/server/modules/job/JobController.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            jobId: job.id,
+          })
+          await this.sqsService.sendMessage(messageParams)
+          return
+        }),
+      )
+      dogstatsd.increment('job.start.success', 1, 1)
+    } catch (e) {
+      logger.error(`error creating and starting job: ${e}`)
+      dogstatsd.increment('job.start.failure', 1, 1)
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/server/modules/job/JobController.ts:53]: will do this in one go for all the other metrics too! good idea

File: src/server/config.ts [hunk 2]
```diff
@@ -173,6 +173,10 @@ export const userAnnouncement = {
 
 export const s3Bucket = process.env.AWS_S3_BUCKET as string
 export const linksToRotate = process.env.ROTATED_LINKS
+export const sqsRegion = process.env.SQS_REGION as string
+export const sqsBulkQRCodeStartUrl =
+  (process.env.SQS_BULK_QRCODE_GENERATE_START_URL as string) || ''
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/server/config.ts:178]: fixed [cc818c3](https://github.com/opengovsg/GoGovSG/pull/2083/commits/cc818c3f62e137799fd3b8c46846c92071f5fe44)

File: README.md [hunk 3]
```diff
@@ -133,7 +133,12 @@ After these have been set up, set the environment variables according to the tab
 |COOKIE_MAX_AGE|Yes|Session duration of cookie|
 |BULK_UPLOAD_MAX_NUM|No|Maximum number of links that can be bulk uploaded at once. Defaults to 1000|
 |BULK_UPLOAD_RANDOM_STR_LENGTH|No|String length of randomly generated shortUrl in bulk upload. Defaults to 8|
+|BULK_QR_CODE_BATCH_SIZE|No|Maximum number of links that can be bulk uploaded at once. Defaults to 100|
```
[the remark this answers is not in the document]
Comment (gweiying) [reply README.md]: fixed [cc818c3](https://github.com/opengovsg/GoGovSG/pull/2083/commits/cc818c3f62e137799fd3b8c46846c92071f5fe44)

**Pull request merged**